scalableminds/webknossos
# Export Volume Annotations

![image](https://user-images.githubusercontent.com/1390035/110470024-a2f31e80-80da-11eb-9525-1c8e96147363.png)



Note that #5252 should be merged first (some of the bugs fixed there could crop up here too during testing)

### TODO (includes changes in https://github.com/scalableminds/webknossos-worker/pull/6/ )
 - [x] fetch volume annotations
 - [x] merge with fallback layer
 - [x] pass result to cuber
 - [x] are compound tracings cached long genough? → increased to 70 min, created follow-up issue #5255 
 - [x] use glob instead of exists
 - [x] pass version
 - [x] pass token only for this job
 - [x] pass tracing id from frontend
 - [x] don’t include fallback layer if it is unlinked
 - [x] test resulting files with imagemagick (`convert 000002.tiff -evaluate multiply 200 -depth 8 8.tif`)
 - [x] adapt job description in list view
 - [x] disable buttons already in frontend if mag1 is not present
 - [x] ensure saved state
 - [x] clean up intermediate files? (let’s leave them in for the moment, for debugging. clean-up job issue #5202)

### Steps to test:
- set up https://github.com/scalableminds/webknossos-worker/pull/6/
- set `jobsEnabled = true` in application.conf
- open volume annotation (also try hybrid, skeleton, view mode)
- brush some, create bbox
- click export tif button, modal should show annotation layer
- zip download from jobs list should include segmentation tifs with the annotations
- modals should not contain typos

### Issues:
- fixes #5203 

------
